certbot/certbot/_internal/main.py

```diff
@@ -498,29 +490,70 @@ def _report_new_cert(config, cert_path, fullchain_path, key_path=None):
 
     """
     if config.dry_run:
-        _report_successful_dry_run(config)
+        display_util.notify("The dry run was successful.")
+        return
+
+    assert cert_path and fullchain_path, "No certificates saved to report."
+
+    display_util.notify(
+        ("\nSuccessfully received certificate.\n"
+        "Certificate is saved at: {cert_path}\n{key_msg}"
+        "This certificate expires on {expiry}.\n"
+        "These files will be updated when the certificate renews.\n{renew_msg}{nl}").format(
+            cert_path=fullchain_path,
+            expiry=crypto_util.notAfter(cert_path).date(),
+            key_msg="Key is saved at:         {}\n".format(key_path) if key_path else "",
+            renew_msg="Certbot will automatically renew this certificate in the background."
+                      if config.preconfigured_renewal else
+                      (f'Run "{cli.cli_constants.cli_command} renew" to renew '
+                       "expiring certificates. "
+                       "We recommend setting up a scheduled task for renewal; see "
+                       "https://certbot.eff.org/docs/using.html#automated-renewals "
+                       "for instructions."),
+            nl="\n" if config.verb == "run" else "" # visually split output if also deploying
+        )
+    )
+
+
+def _csr_report_new_cert(config, cert_path, chain_path, fullchain_path):
+    # type: (interfaces.IConfig, Optional[str], Optional[str], Optional[str]) -> None
+    """ --csr variant of _report_new_cert.
+
+    Until --csr is overhauled (#8332) this is transitional function to report the creation
+    of a new certificate using --csr.
+    TODO: remove this function and just call _report_new_cert when --csr is overhauled.
+
+    :param config: Configuration object
+    :type config: interfaces.IConfig
+
+    :param cert_path: path to cert.pem
```

Comment:
```suggestion
    :param str cert_path: path to cert.pem
```

Reply:
We only need the fancy two-line version when it's a custom object. This applies to the two below as well.